### ISSUE TITLE
Incorrect code generated for custom TypeConverter with Generics

### ISSUE THREAD

**Issue by xsahil03x** [simolus3/moor#3300]
**Version Used**

`drift: 2.21.0`
`drift_dev: 2.21.0`

**Details**

We have a custom TypeConverter for converting to and from `Map` Object.

```dart
/// Maps a [Map] of type [String], [T] into a [String] understood
/// by the sqlite backend.
class MapConverter<T> extends TypeConverter<Map<String, T>, String> {
  @override
  Map<String, T> fromSql(String fromDb) {
    return Map<String, T>.from(jsonDecode(fromDb) ?? {});
  }

  @override
  String toSql(Map<String, T> value) {
    return jsonEncode(value);
  }
}
```

And a sample usage is as follows:
```dart
/// Map of custom user extraData
TextColumn get extraData => text().map(MapConverter<Object?>())();
```

Earlier it used to generate this static converter.
```dart
static TypeConverter<Map<String, Object?>, String> $converterextraData = MapConverter();
```

but with the latest version, It generates it with a type
```dart
static TypeConverter<Map<String, Object?>, String> $converterextraData = MapConverter<Object>();
```

Check the `non-nullable Object` type at the end in the new generated code. Which results in error when the data is null. Ideally it should be `MapConverter<Object?>` as defined in `extraData` field.

```dart
static TypeConverter<Map<String, Object?>, String> $converterextraData = MapConverter<Object?>();
```

**Comment by simolus3** [comment]
Thanks for the report! We've improved handling generic code in 2.20.0 by including the type argument in the generated code, but then I've messed things up by forgetting the optional question mark.

This has been fixed in the referenced commit, I'll also prepare a patch release.

**Comment by simolus3** [comment]
Should be fixed in version 2.21.1 of `drift_dev`
